modules/tinymce/src/plugins/table/main/ts/queries/TableTargets.ts

```diff
@@ -5,53 +5,40 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Element as DomElement, Node } from '@ephox/dom-globals';
 import { Fun, Option } from '@ephox/katamari';
+import { Element } from '@ephox/sugar';
 import * as CellOperations from './CellOperations';
+import { RunOperation } from '@ephox/snooker';
 
-const noMenu = function (cell) {
-  return {
-    element: Fun.constant(cell),
-    mergable: Option.none,
-    unmergable: Option.none,
-    selection: Fun.constant([ cell ])
-  };
-};
-
-const forMenu = function (selections, table, cell) {
-  return {
-    element: Fun.constant(cell),
-    mergable: Fun.constant(CellOperations.mergable(table, selections)),
-    unmergable: Fun.constant(CellOperations.unmergable(cell, selections)),
-    selection: Fun.constant(CellOperations.selection(cell, selections))
-  };
-};
+const noMenu = (cell: Element<DomElement>): RunOperation.CombinedTargets => ({
+  element: Fun.constant(cell),
+  mergable: Option.none,
+  unmergable: Option.none,
+  selection: Fun.constant([ cell ])
+});
 
-const notCell = function (element) {
-  return noMenu(element);
-};
+const forMenu = (selections, table, cell: Element<DomElement>): RunOperation.CombinedTargets => ({
+  element: Fun.constant(cell),
+  mergable: Fun.constant(CellOperations.mergable(table, selections)),
+  unmergable: Fun.constant(CellOperations.unmergable(cell, selections)),
+  selection: Fun.constant(CellOperations.selection(cell, selections))
+});
 
-// TODO: types
-const paste = (element: any, clipboard: any, generators: any) => ({
+const paste = (element: Element<DomElement>, clipboard: Element<Node>, generators: any): RunOperation.TargetPaste => ({
```

Comment:
I checked and the clipboard here should always be `Element<HTMLTableElement>`. See `opPasteCells` in `TableOperations.ts`.

Reply:
Note: To fix this, you probably need to convert the node name check to a type guard in `Clipboard.ts`. eg:
```
const isTable = (elm: Element<DomNode>): elm is Element<HTMLTableElement> => Node.name(elm) === 'table';
```